### ISSUE TITLE
Custom POST body not being sent in v3.1.5

### ISSUE THREAD

**Issue by adamjweaver** [Alamofire/Alamofire#1032]
Hi

This morning I updated Alamofire from 3.1.3 to 3.1.5 and since then my custom POST requests no longer include the POST body. I downgraded back to 3.1.3 and it worked again. I couldn't see anything in the changelog that might explain it so wondered if anyone could shed any light on it. Request creation code as follows:

```
self.manager.request(httpMethod, url, headers: headers, parameters: [:], encoding: .Custom({ (convertible, _) -> (NSMutableURLRequest, NSError?) in
            let mutableRequest = convertible.URLRequest.copy() as! NSMutableURLRequest
            if let j = json {
                mutableRequest.HTTPBody = j.dataUsingEncoding(NSUTF8StringEncoding, allowLossyConversion: false)
            }
            return (mutableRequest, nil)
        }))
```

where `json` is a pre-encoded json string.

Thanks #

**Comment by cnoon** [comment]
Hi @adamjweaver, the reason that it is no longer being called is that we no longer call the `.Custom` closure in the event that there are no parameters being passed in. This was changed in the `3.1.4` release. However, due to several issues being opened on this topic, we're debating as to whether reverse the logic. Please hold tight for now while we figure this out. For now, you'll need to stay on the 3.1.3 release unless you can find another workaround.

We don't intend the `.Custom` encoding to be used unless you actually have parameters to encode. However, we've had multiple reports of users using this for other things than what it was intended.

**Comment by cnoon** [comment]
Okay @adamjweaver, thanks for your patience. We've decided to revert the changes in 75c269ae and instead handle the case of an empty query string properly for URL encoding. I've fixed up the issue in 8b618c76 if you want to check it out. These changes will be available in the 3.2.0 which we are working on right now.

Cheers. 🍻

**Comment by Junus** [comment]
@cnoon Hi! looks like version 3.2.0 doesn't work. When I'm sending .POST request body is empty, but when I'm using .PUT everything ok
```
encoding = .Custom({convertible, params in
                let mutableRequest = convertible.URLRequest.copy() as! NSMutableURLRequest
                mutableRequest.HTTPBody = body
                return (mutableRequest, nil)
            })

````
